workspaces/ui-v2/src/optic-components/hooks/diffs/SharedDiffContext.tsx

```diff
@@ -166,6 +167,22 @@ export const SharedDiffStore = (props: SharedDiffStoreProps) => {
     reset: () => send({ type: 'RESET' }),
     handledCount: [handled, total],
     startedFinalizing: () => send({ type: 'USER_FINISHED_REVIEW' }),
+    setEndpointName: (id: string, command: any) =>
+      send({
+        type: 'SET_ENDPOINT_NAME',
+        id,
+        command,
+      }),
+    setPendingEndpointName: (id: string, name) => {
+      const pendingEndpoint = context.pendingEndpoints.find((i) => i.id === id);
+      if (!pendingEndpoint) {
+        return console.error(`Could not find pending endpoint with id ${id}`);
+      }
+      pendingEndpoint.ref.send({ type: 'STAGED_ENDPOINT_NAME_UPDATED', name });
```

Comment:
Here we delegate to the other state machine which is used to generate the pending endpoint commands - there's better type safety we can add here though. We need to trigger a recompute commands here in the top level xstate store though

Reply:
> there's better type of safety we can add here though

It's been on my maintenance list to figure out how to get the [event / schema typings to pass through](https://xstate.js.org/docs/guides/typescript.html#using-typescript) -- it hasn't been able to infer as much as I was promised. Happy to pair on it sometime

Reply:
I think we need to pass in the `ref` type which is the pendingEndpoint state machine? I'll need to spend some time digging through and understanding the xstate typescript typing though